libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/incremental/ClasspathSnapshot.kt

```diff
@@ -95,10 +94,11 @@ open class ClasspathSnapshot protected constructor(
             }
         }
 
-        // We do not compute structural data for unchanged files of the current snapshot for performance reasons. That is why we
-        // update the previous snapshot as that one contains all entries.
-        computedClasspathData.putAll(previousData)
-        computedClasspathData.putAll(currentData)
+        // We do not compute structural data for unchanged files of the current snapshot for performance reasons.
+        // That is why we reuse the previous snapshot as that one contains all unchanged entries.
+        previousData.filter { !computedClasspathData.containsKey(it.key) }.forEach {
```

Comment:
Minor, style: `filterTo`, destructuring parameters and `contains` operator can be used:
```
previousData.filterTo(computedClasspathData) { (key, _) -> key !in computedClasspathData }
```